shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
I tried injecting a `KJ_FAIL_REQUIRE()` in the `stayAwake()` implementation in supervisor.c++ and I noticed that this `close()` method never actually gets called! Apparently wrapping the capability with `new Capnp.Capability()` before returning it causes this `close()` to be shadowed or something. If I remove that wrapper, `close()` gets called as expected.

(Actually, you don't even need to inject a `KJ_FAIL_REQUIRE()`; `stayWake()` drops the handle in any case.)